Add Support For AWS Dex Basic-Authentication

PatrickXYS opened the pull request:
It's for internal review.

Jeffwan commented on kfdef/source/master/kustomization.yaml:
```diff
@@ -5,6 +5,7 @@ resources:
 - kfctl_anthos.yaml
 - kfctl_aws.yaml
 - kfctl_aws_cognito.yaml
+- kfctl_aws_dex.yaml
```
I think our plan is to fit dex into kfctl_aws.yaml? isn't it?

PatrickXYS replied:
Right, fixed in the new revision.

Jeffwan commented on kfdef/kfctl_aws_dex.v1.0.0.yaml:
```diff
@@ -0,0 +1,426 @@
+apiVersion: kfdef.apps.kubeflow.org/v1
```
Don't need to generate for v1.0.0 - v1.0.2. They are all historical versions

PatrickXYS replied:
Fixed.

jlewi committed:
Remove GCP KFDefs and E2E tests. (#1237)

* Per #1063 with 1.1 GCP will no longer use KFDef to deploy
  Kubeflow; isntead we will be using blueprints kubeflow/gcp-blueprints

  * So we remove all the GCP KFDefs

  * Cleanup the kfdef/OWNERs; remove Googlers since there are no longer
    anymore GCP configs.

* Fix #818 - stop running E2E tests that deploy Kubeflow on every
  commit.

  * We now have unittests aimed at validating the manifests; so it
    no longer makes sense to fully deploy KF and verify its working.

    * E2E tests should now happen downstream.

    * We also need to remove the tests since we are removing GCP KFDefs on
      which they depend.

Jeffwan committed:
Upgrade ingress APIVersion to support k8s 1.16 (#1245)

Bobgy committed:
Mysql kustomize v3 manifest for both generic and gcp stack (#1247)

* Add argo to stacks/generic

* Pull pipelines manifest from upstream

* Updated kfp

* Minio v3 manifests

* Rename minio configmap

* Add generic minio install

* Generate new test data

* Mysql kustomize v3 manifest - generic install

* Add mysql gcp pd install

* Generate test data

andreyvelich committed:
Add Katib to GCP stacks (#1249)

andreyvelich committed:
Katib manifests for Kubeflow 1.1 (#1239)

* Change manifest for Kubeflow 1.1

* Generate tests

* Change image tag from 5df1bd2 to 917164a

Bobgy committed:
Document recommendation of kustomize version (#1240)

* Add documentation of kustomize version

* Update README.md

shawnzhu committed:
Fixes local tests (#1255)

xfate123 committed:
unmatched label for deployment config (#1223)

thesuperzapper committed:
replace `env:` with `envs:` in Kustomization resources (#1204)

kubeflow-bot committed:
[auto PR] Update the pytorch-operator image to tag vmaster-gd596e904 (#1273)

* image gcr.io/kubeflow-images-public/pytorch-operator:vmaster-gd596e904
* Image built from kubeflow/kubeflow@d596e904

Bobgy committed:
Pipelines kustomize v3 manifest (#1248)

* Add argo to stacks/generic

* Pull pipelines manifest from upstream

* Updated kfp

* Minio v3 manifests

* Rename minio configmap

* Add generic minio install

* Generate new test data

* Mysql kustomize v3 manifest - generic install

* Add mysql gcp pd install

* Generate test data

* Pipelines kustomize v3 manifests

* Add kfp ui virtual service

* Add metadata deployment to stacks/generic

* Use common cluster domain

* Deploy metadata writer

* Add kfp cache server

* Update test data

* Updated kfp to rc2

* simplify pipeline v3 manifest using updated kfp rc2 manifest

* Fix pipeline-install-config

* remove redundant configmap

* update tests

* Fix env -> envs transform when pulling kfp upstream manifest

* Pull again

kubeflow-bot committed:
[auto PR] Update the xgboost-operator image to tag vmaster-g8f8c3f96 (#1276)

* image gcr.io/kubeflow-images-public/xgboost-operator:vmaster-g8f8c3f96
* Image built from kubeflow/kubeflow@8f8c3f96

yuzisun committed:
Use ON_WITH_EXCLUSION for istio dex (#1257)

Bobgy committed:
GCP stack (#1286)

* Add gcp stack back

* Fix gcp stack

andreyvelich committed:
add pytorchjobs/finalizers to rbac (#1283)

Tomcli committed:
add namespace selector and test case for admission (#1284)

jlewi committed:
Fix GCP stacks which aren't building. (#1288)

* The GCP stack doesn't build (kubeflow/gcp-blueprints#55). The problem
  is pipeline GCP manifest is referring to a configmap that doesn't
  exist

  * This is a temporary hack to make the manifests build; I just commented
    out the missing file.

  * @bobgy will hopefully add an appropriate long term fix.

* The GCP stack was reintroduced in #1286. The unittests should have
  caught this but they didn't because make generate is swallowing errors
  if there is a problem generating the tests #1269

Update the alice example to use the GCP config.

andreyvelich committed:
OWNERS for TF and PyTorch operators (#1289)

* Add owners for tf and pytorch operators

* Update test owners

kubeflow-bot committed:
[auto PR] Update the tf_operator image to tag vmaster-ga2ae7bff (#1270)

* image gcr.io/kubeflow-images-public/tf_operator:vmaster-ga2ae7bff
* Image built from kubeflow/kubeflow@a2ae7bff

jlewi committed:
Update ingresses and deployments to work with 1.16 (#1287)

* ingress is now networking.k8s.io/v1beta1 this was introduced in
  in 1.8 so there shouldn't be any blocker to upgrading our ingresses now
  https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

  * Most of the ingresses had already been updated

* Deployments are now apps/v1 not extensions/v1beta1

* Delete gcp/basic-auth-ingress rather than updating it.

* Delete kfdef/generic/istio rather than update it

  * It doesn't look like this has been modified since it was copied over
    from the kubeflow/kubeflow repo in 2019. ISTIO configs should be in /istio

* Add a test for deprecated Kubernetes resources fixes #375

Regenerate tests.

jlewi committed:
Follow up to #1288 (#1292)

* Remove commented out code which turns out not to be needed
* Fix OWNERs file

Commit:
Add Support For AWS Dex Basic-Authentication

adrian555 committed:
update kfctl_ibm KfDef to kustomize v3  (#1246)

* update kfctl_ibm kfdef to kustomize v3

* small update to README

* update to use katib, minio and mysql generic

* update after platform test

* fix test failure

krishnadurai committed:
Generates and fixes unit-tests (#1297)

jlewi committed:
Synchronize the tests (#1300)

* Related to #1252

Jeffwan committed:
Fix aws ingress api version bug (#1293)

* Fix ingress api version bug

* Update tf-operator image tag in gcp test_data

Commit:
Fix Registered Issue for aws overlay of Jupyter-web-app
